src/NewTools-Sindarin-Tools/StSindarinDebuggerCreateCommandPresenter.class.st

```diff
@@ -0,0 +1,153 @@
+Class {
+	#name : 'StSindarinDebuggerCreateCommandPresenter',
+	#superclass : 'SpPresenterWithModel',
+	#instVars : [
+		'descriptionInput',
+		'cancelButton',
+		'createButton',
+		'package',
+		'iconList',
+		'iconNameSelected',
+		'commandName',
+		'code',
+		'projectName'
+	],
+	#category : 'NewTools-Sindarin-Tools-Presenters',
+	#package : 'NewTools-Sindarin-Tools',
+	#tag : 'Presenters'
+}
+
+{ #category : 'accessing' }
+StSindarinDebuggerCreateCommandPresenter >> code: aCode [
+	code := aCode 
+]
+
+{ #category : 'accessing' }
+StSindarinDebuggerCreateCommandPresenter >> commandName: aName [
+	commandName := aName asCamelCase 
+]
+
+{ #category : 'initialization' }
+StSindarinDebuggerCreateCommandPresenter >> connectPresenters [ 
+	iconList transmitTo: self
+]
+
+{ #category : 'actions' }
+StSindarinDebuggerCreateCommandPresenter >> createCommand [
+	| name className description class |
+	name := commandName ifNil: ['Default'].
+	className := 'Sindarin{1}Command' format: {name asCamelCase}.
+	description := descriptionInput text ifNil: [''].
+	
+	class := SindarinCommand << className 
+		package: package;
+		build;
+		install.
+
+	class compile: ('execute
+	|sindarin|
+	sindarin := self context sindarinDebugger.
+	self context debuggerActionModel preventUpdatesDuring: [
+		{1}
+	]' format: {code}) classified: 'executing'.
+	
+	class class compile: ('defaultIconName
+	^#{1}' format: {iconNameSelected}) classified: 'initialization'.
+	
+	class class compile: ('defaultDescription
+	^ ''{1}''' format: {description}) classified: 'initialization'.
+	
+	class class compile: ('defaultName
+	^ ''{1}''' format: {name}) classified: 'initialization'.
+	
+	self createMenu.
+
+]
+
+{ #category : 'actions' }
+StSindarinDebuggerCreateCommandPresenter >> createMenu [
+	StDebugger class compile: ('buildSindarin{2}ExtentionCommandsGroupWith: stDebuggerInstance forRoot: rootCommandGroup
+	<extensionCommands>
+	| commands toolbarCommandGroup |
+	commands := (SindarinCommand subclasses select: [ :sub | sub packageName = ''{3}'']) collect: [ :each | each forSpecContext: stDebuggerInstance ].
+	
+	toolbarCommandGroup := CmCommandGroup forSpec
+		beToolbarPopoverButton;
+		name: ''{1}'';
+		icon: (stDebuggerInstance application iconNamed: #changeUpdate);
+		yourself.
+	
+	(self debuggerToolbarCommandGroupFor: rootCommandGroup) register: toolbarCommandGroup.
+	
+	commands do: [ :c | toolbarCommandGroup register: c ].' format: {projectName. projectName asCamelCase . package})
```

Comment:
```suggestion
	commands do: [ :command | toolbarCommandGroup register: command ].' format: {projectName. projectName asCamelCase . package})
```

In Smalltalk we try to avoid abbreviation. Since we have completion tool it's better to have a readable code than to gain 2sec to type the code